bpfman-operator/apis/v1alpha1/cluster_uprobe_program_types.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2023 The bpfman Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// All fields are required unless explicitly marked optional
+// +kubebuilder:validation:Required
+package v1alpha1
+
+// ClUprobeProgramInfo contains the information for the uprobe program
+type ClUprobeProgramInfo struct {
+	// links in the list of points to which the program should be attached.  The list items
+	// are optional and may be udated after the bpf program has been loaded
+	// +optional
+	// +kubebuilder:default:={}
+	Links []ClUprobeAttachInfo `json:"links"`
+}
+
+type ClUprobeAttachInfo struct {
+	// function to attach the uprobe to.
+	// +optional
+	Function string `json:"function"`
+
+	// offset added to the address of the function for uprobe.
+	// +optional
+	// +kubebuilder:default:=0
+	Offset uint64 `json:"offset"`
```

Comment:
is there a reasonable upper bound you can set here?

Reply:
The combination of target, function, and offset is used to define the location of an instruction at which the user wants to invoke the uProbe program.  So, the max offset would be the size of the largest allowable executable or library.  The theoretical limit would depend on a number of factors including things like file format, OS, and file system.  In any case, it would be pretty big, and I don't think it would be very helpful because in most cases it would need to be much smaller in practice.  If the user provides an offset that is out of bounds for the given target, the eBPF verifier will return an error.

Reply:
How would that error manifest to users?

Reply:
1. The BpfApplication CRD would have Status == "Error", and it would list which BpfApplicationState CRD was in error. 
2. The BpfApplicationState CRD in question would have Status == "Error"
3. The failed link would have LinkStatus == "AttachError" in the state CRD

Reply:
What is the impact to user workloads in this state?

Reply:
Other links that don't have errors will still be attached.  It's up to the user application to decide what to do when this happens.  Then, they may also need to figure out why what they are trying to do doesn't work.

eBPF is complicated, and the kernel isn't very helpful in telling you what's wrong when there's an error.  So, we will also log what we can about the error, but often it's just `EINVAL` and the eBPF application developer would need to use their knowledge of eBPF and the Linux kernel to figure out what's wrong.

Reply:
To clarify, by "workload" i'm meaning the non-ebpf applications that are running on a cluster via things like `Deployment`s